controller/config_types.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+import dataconf
+import json
+from typing import Optional, Union, List
+
+
+@dataclass
+class S3Config:
+    """The configuration needed to upload metrics to S3."""
+    endpoint: str
+    bucket: str
+    path_prefix: str
+    access_key_id: str
+    secret_access_key: str
+    rotation_period_seconds: Optional[str] = None
+
+    def __post_init__(self):
+        if not self.rotation_period_seconds:
+            self.rotation_period_seconds = 86400
```

Comment:
You can set the default value above instead with `    rotation_period_seconds: Optional[str] = 86400`